src/SignalR/common/Protocols.Json/src/JsonHubProtocolOptions.cs

```diff
@@ -11,30 +11,9 @@ namespace Microsoft.AspNetCore.SignalR
     /// </summary>
     public class JsonHubProtocolOptions
     {
-        internal readonly JsonSerializerOptions _serializerOptions;
-
-        public JsonHubProtocolOptions()
-        {
-            _serializerOptions = JsonHubProtocol.CreateDefaultSerializerSettings();
-        }
-
-        public bool IgnoreNullValues { get => _serializerOptions.IgnoreNullValues; set => _serializerOptions.IgnoreNullValues = value; }
-        public bool WriteIndented { get => _serializerOptions.WriteIndented; set => _serializerOptions.WriteIndented = value; }
-        public bool AllowTrailingCommas { get => _serializerOptions.AllowTrailingCommas; set => _serializerOptions.AllowTrailingCommas = value; }
-        public bool UseCamelCase
-        {
-            get => _serializerOptions.PropertyNamingPolicy == JsonNamingPolicy.CamelCase;
-            set
-            {
-                if (value)
-                {
-                    _serializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
-                }
-                else
-                {
-                    _serializerOptions.PropertyNamingPolicy = null;
-                }
-            }
-        }
+        /// <summary>
+        /// Gets or sets the settings used to serialize invocation arguments and return values.
+        /// </summary>
+        public JsonSerializerOptions PayloadSerializerSettings { get; set; } = JsonHubProtocol.CreateDefaultSerializerSettings();
```

Comment:
Payload is kinda a wierd word. Not wierd like *walking dan back to building 18 the long-way* wierd, but still a little wierd. You do you if SignalR already uses this term for stuff 😆

Reply:
Also a little strange to see `public FooOptions  FooBar*Settings*` - even though we usually use options for `IOptions<>`-related stuff

Reply:
We do :)
https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/common/Protocols.NewtonsoftJson/src/NewtonsoftJsonHubProtocolOptions.cs#L17

Reply:
I'm generally all for consistency but also happy to re-open this bikeshed a little given the breaking-ness of the protocol change in 3.0.

"Payload" was chosen original because the serializer options **only apply to arguments and return-values** as opposed to applying to the entire protocol packet. I still think that's the most reasonable name but am happy to entertain alternatives :).

As for "Settings", we used that when the type we configured was `JsonSerializerSettings` but now it's `JsonSerializerOptions`. I'm OK with changing that to `Options`.

`PayloadSerializerOptions` makes sense to me if we're keeping Payload around.

Reply:
👍 for `PayloadSerializerOptions` but no biggie.